app/Keycloak/Listeners/CreateMissingClients.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Keycloak\Listeners;
+
+use App\Domain\Integrations\Repositories\IntegrationRepository;
+use App\Keycloak\Jobs\MissingClientsDetected;
+use App\Keycloak\Repositories\KeycloakClientRepository;
+use App\Keycloak\Service\CreateClientForRealms;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+final class CreateMissingClients implements ShouldQueue
```

Comment:
I would bundle the `handle` methods from `CreateClients` and `CreateMissingClients`. As far as I can see the only difference is that `getMissingRealmsByIntegrationId` gets called.

So you can create two public methods `handleCreateClients` and `handleCreatingMissingClients` which both call a private method `handle` that does the common part.